### ISSUE TITLE
Reference Test_0013_norm_i18n_256. When Unicode \u{1F4A9} hit the fan ?

### ISSUE THREAD

**Issue by branciard** [branciard/iscc-core-ts#1]
In [python reference implementation code](https://github.com/iscc/iscc-core/blob/main/iscc_core/code_meta.py#L209) unicode category C is filtered.

Unicode category C "Other"  is composed of subcategory 
- cc = "Control"
- cf = "Format"
- cs = "Surrogate"
- co = "Private_Use"
- cn = "Unassigned"

Unicode \u{1F4A9}  is from category cs according to this [test](https://github.com/branciard/iscc-core-ts/blob/main/src/data.test.ts#L201)  

Test_0013_norm_i18n_256 result is different if you filter or not this unicode in the description. 
Do we have to filter cs catagory or not ?

[Test in Python impl result](https://github.com/iscc/iscc-core/blob/main/iscc_core/data.json#L167) vs [Test in typescript impl result](https://github.com/branciard/iscc-core-ts/blob/main/src/data.test.ts#L217) ?

**Comment by titusz** [comment]
Yes, you need to filter characters from all those categories except for certain newline characters specified in https://github.com/iscc/iscc-core/blob/main/iscc_core/options.py#L58

**Comment by branciard** [comment]
The issue is likely with how JavaScript/TypeScript and Python handle surrogate pairs differently in string length and slicing operations. Let's modify the sliding window function to handle Unicode characters correctly.

sliding_window function to fix

The key changes:
Use Array.from(seq) to split the string into an array of Unicode characters (this handles surrogate pairs correctly)
Use the actual code point length instead of string length (which counts surrogate pairs as two characters in JavaScript)
Use array slicing and join instead of substring (which can split surrogate pairs)
For example:
The string "💩" has:
length of 2 in JavaScript (due to surrogate pair D83D DCA9)
length of 1 in Python (single code point)
Array.from("💩") gives us the correct length of 1

This should make the sliding window behavior match Python's string slicing, which naturally handles Unicode code points correctly.
